src/Polly.Shared/Wrap/PolicyWrap.cs

```diff
@@ -21,12 +21,12 @@ internal PolicyWrap(Action<Action<Context, CancellationToken>, Context, Cancella
     /// <typeparam name="TResult">The return type of delegates which may be executed through the policy.</typeparam>
     public partial class PolicyWrap<TResult> : Policy<TResult>, IPolicyWrap<TResult>
     {
-        internal PolicyWrap(Func<Func<Context, CancellationToken, TResult>, Context, CancellationToken, TResult> policyAction, Policy outer, IsPolicy inner)
+        internal PolicyWrap(Func<Func<Context, CancellationToken, TResult>, Context, CancellationToken, TResult> policyAction, ISyncPolicy outer, ISyncPolicy<TResult> inner)
             : base(policyAction, outer.ExceptionPredicates,  PredicateHelper<TResult>.EmptyResultPredicates)
         {
         }
 
-        internal PolicyWrap(Func<Func<Context, CancellationToken, TResult>, Context, CancellationToken, TResult> policyAction, Policy<TResult> outer, IsPolicy inner)
+        internal PolicyWrap(Func<Func<Context, CancellationToken, TResult>, Context, CancellationToken, TResult> policyAction, ISyncPolicy<TResult> outer, IsPolicy inner)
```

Comment:
Why was the 3rd param left as type IsPolicy? Not consistent with your other changes

Reply:
Good observation, thanks!

Just like in PolicyWrapAsync.cs, there should be 3 variants:
1: ..., ISyncPolicy outer, ISyncPolicy<TResult> inner)
2: ..., ISyncPolicy<TResult> outer, ISyncPolicy inner)
3: ..., ISyncPolicy<TResult> outer, ISyncPolicy<TResult> inner)

I will split up the IsPolicy into 2 and 3, which also aligns it with the changes made to PolicyWrapAsync.cs.

I have pushed the changes to commit b3cc1a7.